### PR TITLE
guess username field by id

### PR DESCRIPTION
**Pull request by felixb**
Guess username fields by id instead of just picking the first one.
The current behavior overrides the account field for the aws console.

### CODE REVIEW

**Comment by maximbaz** [comment]
Related: https://github.com/dannyvankooten/browserpass/pull/78#issuecomment-301589526

Guessing by id is a good idea, but I think that "first of type" should be removed entirely.

File: chrome/script.browserify.js [hunk 1]
```diff
@@ -195,14 +195,19 @@ function fillLoginForm(login) {
 	}
 
 	update(field('input[type=password]'), ${JSON.stringify(login.p)});
-	update(field('input[type=email], input[type=text], input:first-of-type'), ${JSON.stringify(login.u)});
+	// Guess username field
+	var usernameFields = field('input[id=username], input[id=login], input[id=email]');
```
Comment (maximbaz) [inline chrome/script.browserify.js]: why not just pass all arguments here at once?

```
var usernameFields = field('input[id=username], input[id=login], input[id=email], input[type=email], input[type=text]');
```

Comment (felixb) [inline chrome/script.browserify.js]: you'd get the first match, in case of the AWS console, it's still the account field.

Comment (maximbaz) [inline chrome/script.browserify.js]: could you explain that a little? in my example I excluded the `first-of-type` and rearranged fields, so I think the two code snippets will be identical... or not?

**Comment by felixb** [comment]
In my experience, rearranging the filters did not change the order of the outcome. I guess the list of matches is ordered by appearance in the DOM. Maybe I'm mistaken though.

**Comment by felixb** [comment]
I just removed the `input:first-of-type` and fixed the fallback when no `username` field is found.

**Comment by felixb** [comment]
do you need anything from me to merge this change?

**Comment by maximbaz** [comment]
@dannyvankooten ^

**Comment by felixb** [comment]
@dannyvankooten: do you have any opinion on this?

BTW: This fixes #111.

**Comment by maximbaz** [comment]
I rebased and merged your change in https://github.com/dannyvankooten/browserpass/commit/adaa3a7faefc4b4b79e02c4b84c57a610c1c0394

You were right that the order or selectors did not matter, I fixed that separately and it allowed me to simplify your change. Let me know if it works for you - and thanks!

**Comment by felixb** [comment]
Works great!

**Pull request closed without merging**